src/ucp/rma/amo_sw.c

```diff
@@ -65,7 +65,10 @@ ucp_amo_sw_progress(uct_pending_req_t *self, uct_pack_callback_t pack_cb,
     ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct);
     ucs_status_t status;
 
-    ucs_assert(req->flags & UCP_REQUEST_FLAG_PROTO_AMO_PACKED);
+    if (req->send.ep->worker->context->config.ext.proto_enable) {
+        /* assert is valid only when protocols are enabled */
+        ucs_assert(req->flags & UCP_REQUEST_FLAG_PROTO_AMO_PACKED);
```

Comment:
maybe just move the branch/check inside of the assert statement ?

Reply:
> maybe just move the branch/check inside of the assert statement ?

not needed - compiler will exclude whole `if` statement in case of empty action block